libs/Helpers/src/Helpers/QObjectVectorModel.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+#include <QAbstractListModel>
+#include <QQmlEngine>
+
+namespace Status::Helpers {
+
+/// Generic typed QObject provider model
+///
+/// Supports: source model update
+/// \todo rename it to SharedQObjectVectorModel
+template<typename T>
+class QObjectVectorModel final : public QAbstractListModel
```

Comment:
This was a quick solution to duplicating the same model code or bringing the more costly and untyped and unadapted Qt6 solution proposed by @osmaczko. Not sure if this will stay this way. Added todo though.